tutorials/provisional/mosaic.py

```diff
@@ -0,0 +1,297 @@
+<https://github.com/matplotlib/matplotlib/pull/4384>`__ and the
+`patchwork <https://github.com/thomasp85/patchwork>`__ library for R.
+While we do not implement the operator overloading style, we do
+provide a pythonic API for specifying (nested) Axes layouts.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def label_axes(ax_dict):
+    """
+    Helper to set the title on the Axes.
+
+    Parameters
+    ----------
+    ax_dict : Dict[str, Axes]
+        Mapping between the title / label and the Axes.  Set the left
+        title to be equal to the key.
+    """
+    for k, ax in ax_dict.items():
+        ax.set_title(k, loc="left")
+
+
+###############################################################################
+# Express String specification
+# ============================
+#
+# Using single leters we can "draw" the axes we want.  The following
+
+
+layout = """
+AB
+CD
+"""
+
+###############################################################################
+# will give us 4 axes laid out in an 2x2 grid and generates the
+# same figure layout  as``fig.subplots(2, 2)``.
+
+fig = plt.figure(constrained_layout=True)
+axd = fig.subplot_grid(layout)
+label_axes(axd)
+
+###############################################################################
+# A key difference between `.Figure.subplot_grid` and
+# `.Figure.subplots` is that The returned value from
+# `.Figure.subplot_grid` return a dictionary mapping the labels to the
+# `.axes.Axes` instances created
+
+print(axd)
+
+###############################################################################
+# instead of an array.  For comparison we could do:
+
+fig = plt.figure(constrained_layout=True)
+ax_array = fig.subplots(2, 2, squeeze=False)
+label_axes({str((j, k)): ax
+            for j, row in enumerate(ax_array)
+            for k, ax in enumerate(row)})
+
+###############################################################################
+# If we wanted to plot to the upper left axes we would do either
+
+upper_left = axd['A']
+upper_left = ax_array[0, 0]
+
+
+###############################################################################
+# Something else we can easily do with `.Figure.subplot_grid` is
+# specify a a span.  If we want to add an axes to the right side of
+# our Figure that spans the full vertical space we could do
+
+axd = plt.figure(constrained_layout=True).subplot_grid(
+    """
+    ABE
+    CDE
+    """
+)
+label_axes(axd)
+
+###############################################################################
+# If we do not want to fill in all the spaces in the Figure with axes
+# you can specify it to be black
+
+
+axd = plt.figure(constrained_layout=True).subplot_grid(
+    """
+a-c
+BBB
+-d-
+"""
+)
+label_axes(axd)
+
+
+###############################################################################
+# If you would rather use another letter (rather than a dash ``'-'``)
+# to mark the empty space use *empty_sentinel* to specify the
```

Comment:
```suggestion
# If we prefer to use another letter (rather than a dash ``'-'``)
# to mark the empty space, we can use *empty_sentinel* to specify the
```